### PR TITLE
feat: add weblink to issue (#446)

### PR DESCRIPTION
**Pull request by Syd7**
PR adds the possibility to create a weblink for an issue. #446

### CODE REVIEW

File: internal/cmd/issue/remotelink/remotelink.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+const (
+	helpText = `Connects an issue to a web link.`
+	examples = `$ jira issue weblink ISSUE-1 http://weblink.com weblink-title`
+)
+
+// NewCmdWeblink is a link command.
+func NewCmdWeblink() *cobra.Command {
+	cmd := cobra.Command{
+		Use:     "remotelink ISSUE_KEY WEBLINK_URL WEBLINK_TITLE",
+		Short:   "Link an issue to a weblink",
```
Comment (ankitpokhrel) [inline internal/cmd/issue/remotelink/remotelink.go:22]:
```suggestion
		Short:   "Adds a remote web link to an issue",
```

File: internal/cmd/issue/remotelink/remotelink.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	"github.com/spf13/viper"
+
+	"github.com/ankitpokhrel/jira-cli/api"
+	"github.com/ankitpokhrel/jira-cli/internal/cmdutil"
+	"github.com/ankitpokhrel/jira-cli/internal/query"
+	"github.com/ankitpokhrel/jira-cli/pkg/jira"
+)
+
+const (
+	helpText = `Connects an issue to a web link.`
```
Comment (ankitpokhrel) [inline internal/cmd/issue/remotelink/remotelink.go:14]:
```suggestion
	helpText = 'Remote connects an issue to a remote web link.'
```

File: internal/cmd/issue/remotelink/remotelink.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	examples = `$ jira issue weblink ISSUE-1 http://weblink.com weblink-title`
+)
+
+// NewCmdWeblink is a link command.
+func NewCmdWeblink() *cobra.Command {
+	cmd := cobra.Command{
+		Use:     "remotelink ISSUE_KEY WEBLINK_URL WEBLINK_TITLE",
```
Comment (ankitpokhrel) [inline internal/cmd/issue/remotelink/remotelink.go:21]:
```suggestion
		Use:     "remote ISSUE_KEY WEBLINK_URL WEBLINK_TITLE",
```

File: internal/cmd/issue/remotelink/remotelink.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+		Aliases: []string{"rmln"},
+		Annotations: map[string]string{
+			"help:args": "ISSUE_KEY\tIssue key, eg: ISSUE-1\n" +
+				"WEBLINK_URL\tUrl of the weblink\n" +
+				"WEBLINK_TITLE\tTitle of the weblink",
+		},
+		Run: weblink,
+	}
+
+	cmd.Flags().Bool("web", false, "Open issue in web browser after successful linking")
```
Comment (ankitpokhrel) [inline internal/cmd/issue/remotelink/remotelink.go:34]:
```suggestion
	cmd.PersistentFlags().Bool("web", false, "Open issue in web browser after successful linking")
```

File: internal/cmd/issue/remotelink/remotelink.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	"github.com/ankitpokhrel/jira-cli/pkg/jira"
+)
+
+const (
+	helpText = `Connects an issue to a web link.`
+	examples = `$ jira issue weblink ISSUE-1 http://weblink.com weblink-title`
+)
+
+// NewCmdWeblink is a link command.
+func NewCmdWeblink() *cobra.Command {
```
Comment (ankitpokhrel) [inline internal/cmd/issue/remotelink/remotelink.go:19]:
```suggestion
func NewCmdRemoteLink() *cobra.Command {
```

**Review by ankitpokhrel — COMMENTED** [review]
Hi @Syd7, thank you for working on this 🎉. 

The functionality seems to be working, however some comments needs to be addressed in order for this to get merged.

1. Instead of a separate command, I think this should be a subcommand for `link`. We can add a subcommand called `remote` so that we can do the following. 
   ```go
   $ jira issue link remote KEY-123 https://example.com "Example text"
   ```
2. In current implementation, if you run `jira issue remotelink` it directly calls the Jira API. It would be nice to ask for the required parameters in tui like we do for `jira issue link`.
3. Instead of calling it `WebLink`, let's use `RemoteLink` in the API implementation since the Jira API calls it `RemoteLink`. We can use the term web link in the help text and doc if that's confusing.
4. Also, please update the README.

**Comment by Syd7** [comment]
Hi @ankitpokhrel 👋 thanks for the review. I think I addressed the comments that you pointed out. please let me know if I missed anything.

File: internal/cmd/issue/link/remote/remote.go [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+func (lc *linkCmd) setRemoteLinkTitle() error {
+	if lc.params.title != "" {
+		return nil
+	}
+
+	var ans string
+
+	qs := &survey.Question{
+		Name:     "remoteLinkTitle",
+		Prompt:   &survey.Input{Message: "Remote link Title"},
```
Comment (ankitpokhrel) [inline internal/cmd/issue/link/remote/remote.go:155]: Nitpick:
```suggestion
		Prompt:   &survey.Input{Message: "Remote link title"},
```

File: README.md [hunk 7]
```diff
@@ -455,6 +455,17 @@ $ jira issue link
 $ jira issue link ISSUE-1 ISSUE-2 Blocks
 ```
 
+##### Remote
+The `remote` command lets you add a remote web link to an issue.
+
+```sh
+# adds a remote web link using an interactive prompt
```
Comment (ankitpokhrel) [inline README.md:462]: Nitpick for consistency
```suggestion
# Adds a remote web link using an interactive prompt
```

**Review by ankitpokhrel — COMMENTED** [review]
Hi @Syd7, I have some minor comments, otherwise this looks good. Great work!

**Comment by Syd7** [comment]
Hi again @ankitpokhrel, I resolved the remaining comments let me know how it looks.

**Review by ankitpokhrel — APPROVED** [review]
Thank you @Syd7 🎉

**Pull request merged**